rest/admin_api.go

```diff
@@ -260,17 +260,27 @@ func readReplicationParametersFromJSON(jsonData []byte) (params sgreplicate.Repl
 				err = base.HTTPErrorf(http.StatusBadRequest, "/_replicate sync_gateway/bychannel filter; Missing query_params")
 				return
 			}
-			var channels []string
-			err := json.Unmarshal([]byte(in.QueryParams), &channels)
-			if err != nil {
+
+			//The Channels must be passed as an array of strings
+			if chanarray, ok := in.QueryParams.([]interface{}); ok {
```

Comment:
If it's actually an array of strings, does casting to []string here work?  

As discussed, should we first try casting to []string as you're doing, then fall back to looking for a JSON object like {"channels":["A","B"]}?